src/money_fmt_plug.c

```diff
@@ -0,0 +1,313 @@
+/*
+ * JtR format to crack password protected MS Money files.
+ *
+ * This software is Copyright (c) 2017, Dhiru Kholia <kholia at kth.se> and it
+ * is hereby released to the general public under the following terms:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted.
+ *
+ * All credit goes to Vladimir Berezniker, "Jackcess Encrypt" and Jackcess
+ * projects.
+ */
+
+#if FMT_EXTERNS_H
+extern struct fmt_main fmt_money;
+#elif FMT_REGISTERS_H
+john_register_one(&fmt_money);
+#else
+
+#include <string.h>
+#include <ctype.h>
+#ifdef _OPENMP
+#include <omp.h>
+#ifndef OMP_SCALE
+#define OMP_SCALE               128
+#endif
+#endif
+
+#include "arch.h"
+#include "misc.h"
+#include "common.h"
+#include "formats.h"
+#include "params.h"
+#include "options.h"
+#include "sha.h"
+#include "md5.h"
+#include "rc4.h"
+#include "jumbo.h"
+#include "unicode.h"
+#include "memdbg.h"
+
+#define FORMAT_NAME             "Microsoft Money (2002 to Money Plus)"
+#define FORMAT_LABEL            "money"
+#define FORMAT_TAG              "$money$"
+#define TAG_LENGTH              (sizeof(FORMAT_TAG) - 1)
+#define ALGORITHM_NAME          "MD5/SHA1 32/" ARCH_BITS_STR
+#define BENCHMARK_COMMENT       ""
+#define BENCHMARK_LENGTH        -1001
+#define BINARY_SIZE             0
+#define BINARY_ALIGN            1
+#define SALT_SIZE               sizeof(struct custom_salt)
+#define SALT_ALIGN              sizeof(uint32_t)
+#define PLAINTEXT_LENGTH        20
+#define MIN_KEYS_PER_CRYPT      1
+#define MAX_KEYS_PER_CRYPT      1
+
+#define PASSWORD_DIGEST_LENGTH  16
+#define PASSWORD_LENGTH         40  // UTF16-LE length
+
+static struct fmt_tests money_tests[] = {
+	{"$money$1*fdb51efac3f4e440*bdaa2deb", "openwall"}, // MS Money Plus Sunset Deluxe
+	{"$money$1*73cb979632c4e340*d9de64ed", "Test12345"},
+	{"$money$1*3dc3fc1cc8f4e440*10f74e1b", "12345678901234567890"}, // Max. password length, original password was 1234567890123456789012345
+	// {"$money$1*103c3428c8f4e440*5aa4a678", "Müller"}, // Not yet working, Jackcess does not work either!
+	{"$money$1*103c3428c8f4e440*5aa4a678", "M|ller"}, // Same hash but this password is cracked and accepted in MS Money Plus just fine!
+	{"$money$1*d68792c7c8f4e440*e1c294bf", "|"}, // Original password is "ü", "|" works fine in official software!
+	{"$money$1*353dc5f8c8f4e440*fb840c22", "D"}, // Original password is "ä"
+	{"$money$1*e5eee9fec8f4e440*8f7777c9", "V"}, // Original password is "ö"
+	{"$money$1*8c83d504c9f4e440*1a1108fa", "#"}, // Original password is "£"
+	{"$money$0*9e3eee5bcbf4e440*76a3a059", "openwall"}, // MS Money 2004
```

Comment:
So CP437 is the "OEM codepage". Then we have CP1252 as "ANSI codepage".

I have figured it out:

Letter | hex (CP1252) | relation
------ | ------------ | --------
Ä | 0xc4
D | 0x44 | 0xc4 & 0x7f == 0x44
Ü | 0xdc
\| | 0x7c | 0xdc & 0x7f == 0x7c
Ö | 0xd6
V | 0x56 | 0xd6 & 0x7f == 0x56
£ | 0xa3
\# | 0x23 | 0xa3 & 0x7f == 0x23

So what it does is use an "ANSI" code page, convert to upper and *then* strip 8th bit. I think that means we should set FMT_UTF8 but not FMT_8_BIT, and emit a warning if target encoding is UTF-8. I'll verify that and take care of this within the next days.